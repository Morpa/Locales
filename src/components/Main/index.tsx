import { useCallback } from 'react'
import { useRouter } from 'next/router'

import * as S from './styles'

export type MainProps = {
  title: string
  description: string
  info: string
  url: string
  altText: string
  hero: string
}

const Main = ({ title, description, info, url, altText, hero }: MainProps) => {
  const router = useRouter()

  const handleChangeLang = useCallback(
    (lang: string) => {
      const locale = lang
      router.push(router.pathname, router.asPath, { locale })
    },
    [router]
  )

  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration src={hero} alt={altText} />

      <S.FlagsWrapper>
        <S.Flag onClick={() => handleChangeLang('pt')}>ğŸ‡§ğŸ‡·</S.Flag>
        <S.Flag onClick={() => handleChangeLang('es')}>ğŸ‡ªğŸ‡¸</S.Flag>
        <S.Flag onClick={() => handleChangeLang('en')}>ğŸ‡ºğŸ‡¸</S.Flag>
      </S.FlagsWrapper>

      <S.InfoWrapper>
        <p>{info}</p>
        <p>{url}</p>
      </S.InfoWrapper>
    </S.Wrapper>
  )
}

export default Main
