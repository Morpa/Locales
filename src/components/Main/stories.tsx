import { Story, Meta } from '@storybook/react/types-6-0'
import Main, { MainProps } from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Página en español',
    descripiont: 'Estas leyendo en español',
    info: 'Puede acceder a la página directamente desde la dirección:',
    url: 'https://locales.vercel.app/es',
    altText: 'Un desarrollador frente a una pantalla con código.',
    hero: '/img/spn.svg'
  }
} as Meta

export const Default: Story<MainProps> = (args) => <Main {...args} />
