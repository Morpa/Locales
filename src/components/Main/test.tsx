import { render, screen } from '@testing-library/react'

import Main from '.'

const props = {
  title: 'Título',
  description: 'Descrição'
}

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: props.description })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main {...props} />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
