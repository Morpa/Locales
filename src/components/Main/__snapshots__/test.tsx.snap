// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  padding: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
}

.c2 {
  margin-top: 2rem;
  font-size: 2rem;
  font-weight: 400;
}

.c3 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
}

.c5 {
  cursor: pointer;
  background: transparent;
  border: 0;
  height: 5rem;
  width: 5rem;
  font-size: 3rem;
}

.c6 {
  font-size: 2rem;
}

.c6 p:last-of-type {
  margin-top: 1rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    TÃ­tulo
  </h1>
  <h2
    class="c2"
  >
    DescriÃ§Ã£o
  </h2>
  <img
    class="c3"
  />
  <div
    class="c4"
  >
    <button
      class="c5"
    >
      ðŸ‡§ðŸ‡·
    </button>
    <button
      class="c5"
    >
      ðŸ‡ªðŸ‡¸
    </button>
    <button
      class="c5"
    >
      ðŸ‡ºðŸ‡¸
    </button>
  </div>
  <div
    class="c6"
  >
    <p>
      Info
    </p>
    <p>
      /url
    </p>
  </div>
</main>
`;
